src/hooks/useUserOnboardingContext.ts

```diff
@@ -82,6 +83,22 @@ function useUserOnboardingContextValue<T>(defaultValue: T, callback: (cli: Matri
     return value;
 }
 
+function useShowNotificationsPrompt(): boolean {
+    const [value, setValue] = useState<boolean>(Notifier.shouldShowPrompt());
+    useEffect(() => {
+        window.addEventListener("storage", (event) => {
+            if (event.storageArea === window.localStorage && event.key === "notifications_hidden") {
+                setValue(Notifier.shouldShowPrompt());
+            }
+        });
```

Comment:
Feels like this might be worth putting as an event emit in Notifier or maybe a dispatch? Probably should at least remove the event listener?